plugin/api.go

```diff
@@ -48,6 +48,16 @@ type API interface {
 	// UpdateUser updates a user.
 	UpdateUser(user *model.User) (*model.User, *model.AppError)
 
+	// GetUserStatus will get a user's status.
+	GetUserStatus(userId string) (*model.Status, *model.AppError)
+
+	// GetUserStatusesByIds will return a list of user statuses based on the provided slice of user IDs.
+	GetUserStatusesByIds(userIds []string) ([]*model.Status, *model.AppError)
+
+	// UpdateUserStatus will set a user's status until the user, or another integration/plugin, sets it back to online.
```

Comment:
`until ... sets it back to online`: is the transition to `online` the only state that will override this user status? Is there something about this API call that makes it "sticky" compare to the user organically going to `away`?

Also, there's some context to this that's important, because right now it sounds like, `Change X until something else changes it`, which seems obvious. Is the intent to call out that the user might trigger going back to online without an explicit action (e.g. focussing the app) or some such?

Reply:
Basically what happens if you set the user's status with this API it will be "sticky" as in it won't change with the normal flow from the user's activity until it's set back to online. This is how the REST API endpoint works as well. That make sense?

Maybe the clarification isn't needed but I was trying to point out that if you set the user to away they won't automatically get set back to online when they click back on the app or post a message, etc. It's "stuck" until something (user clicks set me online, plugin or integration does it, etc.) sets it back to online manually

Reply:
Ah, ok, I was getting that impression, but I wasn't certain about the 'stickiness'. If the REST API works that way, it will hopefully be familiar :)